pkg/generate/app/cmd/newapp_test.go

```diff
@@ -674,3 +675,63 @@ func TestRunBuild(t *testing.T) {
 		}
 	}
 }
+
+func TestEnsureValidUniqueName(t *testing.T) {
+	chars := []byte("abcdefghijk")
+	longBytes := []byte{}
+	for i := 0; i < (util.DNS1123SubdomainMaxLength + 20); i++ {
+		longBytes = append(longBytes, chars[i%len(chars)])
+	}
+	longName := string(longBytes)
+	tests := []struct {
+		input       []string
```

Comment:
Mind adding a `name string` field and pass it as the first argument in every `t.Errorf`? Right now if a test fails we either have to derive which one is it from the given inputs or worst case we won't be able to find out.